### PR TITLE
Fix crash on `percent_rank`

### PR DESCRIPTION
**Pull request by lgbo-ustc**
### Changelog category (leave one):

- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Fix  crash  on `percent_rank`. `percent_rank`'s default frame type is changed to `range unbounded preceding and unbounded following`. `IWindowFunction`'s default window frame is considered and now window functions without window frame definition in sql can be put into different `WindowTransfomer`s properly.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by lgbo-ustc** [comment]
fix issue #67594

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit cfeafb1bf062012902be349d00979bfdd437faf7 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/67661/cfeafb1bf062012902be349d00979bfdd437faf7/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/67661/cfeafb1bf062012902be349d00979bfdd437faf7/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by alexey-milovidov** [comment]
1. Tests have failed.
2. Old tests should not be modified. When adding a new feature, add a new test for it.

**Comment by lgbo-ustc** [comment]
@alexey-milovidov The tests have passed

**Review by novikd — COMMENTED** [review]
Also, it'd be nice to see a perf test for this function in a separate PR.

File: src/Planner/PlannerWindowFunctions.cpp [hunk 1]
```diff
@@ -8,41 +10,49 @@
 
 #include <Interpreters/Context.h>
 
-#include <Planner/PlannerSorting.h>
 #include <Planner/PlannerActionsVisitor.h>
+#include <Planner/PlannerSorting.h>
+#include <Planner/Utils.h>
+#include "Utils.h"
```
Comment (novikd) [inline src/Planner/PlannerWindowFunctions.cpp:16]: Seems redundant
```suggestion
```

File: src/Planner/PlannerWindowFunctions.cpp [hunk 2]
```diff
@@ -8,41 +10,49 @@
...
 
 namespace DB
 {
 
 namespace ErrorCodes
 {
-    extern const int NOT_IMPLEMENTED;
+extern const int NOT_IMPLEMENTED;
```
Comment (novikd) [inline src/Planner/PlannerWindowFunctions.cpp:23]: This also can be left as it was for consistency with the codebase.

**Review by novikd — COMMENTED** [review]
I'd make this changes myself but I can't modify the PR branch.

**Pull request merged**